### DISCUSSION TITLE
When desktop runtime isn't installed, is there a method to customize the error message without using an installer?

### DISCUSSION THREAD

**Discussion by xinding1121** [dotnet/wpf#6338] [Q&A]
I have a WPF application that needs .NET 6 Desktop Runtime installed on user's device. When the desktop runtime is missing, the error alert dialog not always shows the specific runtime information which I expect to be '.NET Desktop Runtime 6'. Seems that behaviors vary on different devices due to OS versions perhaps.
We have demos like:
![image](https://user-images.githubusercontent.com/98146379/161231602-fd6ff3d9-7269-4ab2-bc92-08ea460fad0c.png)
![image](https://user-images.githubusercontent.com/98146379/161231821-a17a6602-f125-4f5c-96d5-1b6906078003.png)
![image](https://user-images.githubusercontent.com/98146379/161231559-05f02d51-4594-478e-bb07-57306951d9ad.png)

The last one is what we expect. Is there any method to customize the message and always show the specific runtime information? Any advice or redirection will be appreciated.

**Comment by singhashish-wpf** [comment]
As i understand, this dialog is shown from the dotnet/runtime codebase, not from wpf side. It will be better to ask this question in [@dotnet/runtime](https://github.com/dotnet/runtime) repo.

**Reply by xinding1121** [reply]
Thanks. I'll also leave it to @dotnet/runtime repo.

**Comment by lindexi** [comment]
@xinding1121 See https://github.com/dotnet-campus/dotnetCampus.AppHost/

The Chinese blog: https://blog.walterlv.com/post/share-self-deployed-dotnet-runtime-among-multiple-exes and you can use Google Translate to translate it to your native language

```xml
 <PropertyGroup>
     <AppHostCoreLibMissingDialogMessage>To run this application, you need to xxxxx</AppHostCoreLibMissingDialogMessage>
     <AppHostCoreLibMissingDialogUrl>The url which will be navigated when yes button clicked</AppHostCoreLibMissingDialogUrl>
 </PropertyGroup>

      <ItemGroup>
        <PackageReference Include="dotnetCampus.AppHost" Version="1.0.0-alpha04" />
      </ItemGroup>
```
